crates/ssr/src/lib.rs

```diff
@@ -327,7 +327,10 @@ impl SsrMatches {
 
 impl Match {
     pub fn matched_text(&self) -> String {
-        self.matched_node.text().to_string()
+        // FIXME: There is probably a way to get (the text of) a SyntaxNode with a TextRange?
+        let mut s = self.matched_node.text().to_string();
```

Comment:
Is there a `self.matched_node.text().slice(self.range)`?

Reply:
There is, don't know how I missed that obvious function in `SyntaxText`...

However, `self.range` is where the match resides within the file, which needs to be moved to the beginning (`.start() = 0`) to sit within `self.matched_node`. Easily fixed, but will need to change with partial statement list matching as the beginning can be updated as well. There I'll likely store a separate member with the range _within_ (relative to) `self.matched_node`.

Reply:
Yeah, I guess that's fine. But there's also `TextRange::checked_sub`, if you ever need it.

Reply:
Yeah there are quite a few functions on `TextRange`, but I'm missing add/sub (just `add` with a negative number would be fine) on the start _or_ end. `checked_add/sub` slide the `TextRange` over the window but ie. optional semicolons need a `-1` on just the `end`.